.vitepress/theme/components/LoadingIcon.vue

```diff
@@ -0,0 +1,56 @@
+<script setup lang="ts">
+defineProps<{
+  className?: string;
+  hidden?: boolean;
+  size?: number;
+}>();
+</script>
+
+<template>
+  <span
+    :class="['root', className]"
+    :aria-hidden="hidden"
+    :style="{ '--size': `clamp(${size || 18}px, 8px, 100px)` }"
```

Comment:
I brought a Loading icon from the stackblitz repo and this is my best attempt to convert it for Vue.js 👀